commands/src/main/java/org/wildfly/extras/creaper/commands/auditlog/AuditLogging.java

```diff
@@ -0,0 +1,132 @@
+        return AuditLogDestinationHandler.INSTANCE;
+    }
+
+    public static AuditLogHandler auditLogHandler() {
+        return AuditLogHandler.INSTANCE;
+    }
+
+// ---
+    /**
+     * Configures the destinations where should be the auditable events sent to.
+     */
+    public static final class AuditLogDestinationHandler {
+
+        private static final AuditLogDestinationHandler INSTANCE = new AuditLogDestinationHandler();
+
+        private AuditLogDestinationHandler() {}
+
+        public AuditLogFileHandler file() {
+            return AuditLogFileHandler.INSTANCE;
+        }
+
+        public AuditLogUdpSyslogHandler udpSyslog() {
+            return AuditLogUdpSyslogHandler.INSTANCE;
+        }
+
+        public AuditLogTcpSyslogHandler tcpSyslog() {
+            return AuditLogTcpSyslogHandler.INSTANCE;
+        }
+
+        public AuditLogTlsSyslogHandler tlsSyslog() {
+            return AuditLogTlsSyslogHandler.INSTANCE;
+        }
+    }
+
+    /**
+     * Sets the destination where should be auditable events sent to.
+     */
+    public static final class AuditLogHandler {
+
+        private static final AuditLogHandler INSTANCE = new AuditLogHandler();
+
+        private AuditLogHandler() {}
+
+        public AddAuditLogHandler add(String name) {
+            return new AddAuditLogHandler(name);
+        }
+
+        public RemoveAuditLogHandler remove(String name) {
+            return new RemoveAuditLogHandler(name);
+        }
+    }
+
+// ---
+    public static final class AuditLogFileHandler {
+
+        private static final AuditLogFileHandler INSTANCE = new AuditLogFileHandler();
+
+        private AuditLogFileHandler() {}
+
+        public AddAuditLogFileHandler.Builder add(String name, String path, String formatter) {
+            return new AddAuditLogFileHandler.Builder(name)
+                    .path(path)
+                    .formatter(formatter);
+        }
+
+        public RemoveAuditLogFileHandler remove(String name) {
+            return new RemoveAuditLogFileHandler(name);
+        }
+    }
+
+    public static final class AuditLogUdpSyslogHandler {
+
+        private static final AuditLogUdpSyslogHandler INSTANCE = new AuditLogUdpSyslogHandler();
+
+        private AuditLogUdpSyslogHandler() {}
+
+        public AddAuditLogUdpSyslogHandler.Builder add(String name, String formatter) {
+            return new AddAuditLogUdpSyslogHandler.Builder(name)
+                    .formatter(formatter);
+        }
+
+        public RemoveAuditLogSyslogHandler remove(String name) {
+            return new RemoveAuditLogSyslogHandler(name);
+        }
+    }
+
+    public static final class AuditLogTcpSyslogHandler {
+
+        private static final AuditLogTcpSyslogHandler INSTANCE = new AuditLogTcpSyslogHandler();
+
+        private AuditLogTcpSyslogHandler() {}
+
+        public AddAuditLogTcpSyslogHandler.Builder add(String name, String formatter) {
```

Comment:
Same as above -- if the builder's constructor only has one parameter, this method shouldn't have more.